diff --git a/cpan/Socket/Socket.xs b/cpan/Socket/Socket.xs
index e46c93e..eddefc8 100644
--- a/cpan/Socket/Socket.xs
+++ b/cpan/Socket/Socket.xs
@@ -111,9 +111,11 @@ static int inet_pton(int af, const char *src, void *dst)
     case AF_INET:
       *(struct in_addr *)dst = ((struct sockaddr_in *)&ss)->sin_addr;
       return 1;
+#ifdef AF_INET6
     case AF_INET6:
       *(struct in6_addr *)dst = ((struct sockaddr_in6 *)&ss)->sin6_addr;
       return 1;
+#endif
     default:
       WSASetLastError(WSAEAFNOSUPPORT);
       return -1;
@@ -132,9 +134,11 @@ static const char *inet_ntop(int af, const void *src, char *dst, socklen_t size)
     case AF_INET:
       ((struct sockaddr_in *)&ss)->sin_addr = *(struct in_addr *)src;
       break;
+#ifdef AF_INET6
     case AF_INET6:
       ((struct sockaddr_in6 *)&ss)->sin6_addr = *(struct in6_addr *)src;
       break;
+#endif
     default:
       return NULL;
   }
@@ -150,6 +154,8 @@ static const char *inet_ntop(int af, const void *src, char *dst, socklen_t size)
 #define HAS_INETNTOP
 #endif
 
+#define INET6_ADDRSTRLEN 65
+
 #ifdef NETWARE
 NETDB_DEFINE_CONTEXT
 NETINET_DEFINE_CONTEXT